rust/src/checkpoints.rs

```diff
@@ -293,25 +293,38 @@ fn checkpoint_add_from_state(
     Ok(v)
 }
 
-fn typed_partition_value_from_string(
-    string_value: &str,
+fn typed_partition_value_from_option_string(
```

Comment:
perhaps we can simplify this by wrapping the original `typed_partition_value_from_string` implementation in `typed_partition_value_from_option_string` like so:

```rust
match string_value {
    Some(s) => if s.is_empty() {
        Ok(Value::Null)  // empty string should be deserialized as null
    } else {
        typed_partition_value_from_string(s, date_type)
    },
    None => Ok(Value::Null),
}
```

Reply:
Modified.